### PR TITLE
Normative: Fix Duration.prototype.total() return value

### PR DESCRIPTION
**Pull request by Aditi-1400**
Fixes #2079

### CODE REVIEW

File: spec/duration.html [hunk 1]
```diff
@@ -1187,11 +1190,38 @@ <h1>
...
+          _microseconds_: an integer,
+          _nanoseconds_: an integer,
+          _largestUnit_: a String,
+          optional _relativeTo_: *undefined*, a Temporal.PlainDate, or a Temporal.ZonedDateTime,
+          ): either a normal completion containing either Time Duration Record if there are no infinite values, ~positive overflow~, or ~negative overflow~ in case of infinite values, or an abrupt completion
       </h1>
       <dl class="header">
         <dt>description</dt>
-        <dd>It converts the time units of a duration into a form where lower units are converted into higher units as much as possible, up to _largestUnit_, and returns the result as a Time Duration Record.</dd>
+        <dd>It converts the time units of a duration into a form where lower units are converted into higher units as much as possible, up to _largestUnit_. This function can be used to balance durations with possible infinite values, unlike BalanceDuration which only works for finite values.</dd>
```
Comment (gibson042) [inline spec/duration.html:1224]:
```suggestion
        <dd>It converts the time units of a duration into a form where lower units are converted into higher units as much as possible, up to _largestUnit_. If the Number value for any unit is infinite, it returns a special value indicating the direction of overflow.</dd>
```

File: spec/duration.html [hunk 2]
```diff
@@ -1187,11 +1190,38 @@ <h1>
           _nanoseconds_: an integer,
           _largestUnit_: a String,
           optional _relativeTo_: *undefined*, a Temporal.PlainDate, or a Temporal.ZonedDateTime,
-        )
+        ): either a normal completion containing a Time Duration Record, or an abrupt completion
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>It converts the time units of a duration into a form where lower units are converted into higher units as much as possible, up to _largestUnit_.</dd>
```
Comment (gibson042) [inline spec/duration.html:1197]:
```suggestion
        <dd>It converts the time units of a duration into a form where lower units are converted into higher units as much as possible, up to _largestUnit_. If the Number value for any unit is infinite, it returns abruptly with a *RangeError*.</dd>
```

**Review by gibson042 — APPROVED** [review]
I like this approach, and all existing BalanceDuration call sites either tolerate the RangeError or are guaranteed not to encounter it.

I left a few editorial suggestions around naming and phrasing.

**Review by ptomato — APPROVED** [review]

**Comment by ptomato** [comment]
Marking as draft until presented at TC39. (The conflict seems to be due to the test262 update which maybe got in here accidentally and can be removed?)

**Comment by ptomato** [comment]
This achieved consensus at the July 2022 TC39 plenary meeting. I just merged a test for it into test262: https://github.com/tc39/test262/pull/3627

**Pull request merged**